modules/internal/ChapelDistribution.chpl

```diff
@@ -785,8 +803,11 @@ module ChapelDistribution {
     delete dist;
   }
 
-  proc _delete_dom(dom:BaseDom, param privatized:bool) {
-    dom.dsiDestroyDom();
+  proc _delete_dom(dom, param privatized:bool) {
+    use Reflection;
+    if canResolveMethod(dom, "dsiDestroyDom") {
+      dom.dsiDestroyDom();
```

Comment:
This is a (hopefully) temporary workaround for a bug with recursive iterators. I will add a comment clarifying that and pointing readers to a GitHub issue.